### PR TITLE
182 add require statement

### LINKED ISSUE #182: Add require statement to mitigate blocking proposal
The pool and the voting app pair were designed with the assumption that the voteTime of Api3Voting.sol and EPOCH_LENGTH of StateUtils.sol  would be set to 1 week.

We should require() this (here)[https://github.com/api3dao/api3-dao/blob/5acce3aaddee49934b26b91aa2a4bafd5f175638/packages/api3-voting/contracts/Api3Voting.sol#L99] (i.e., require(voteTime == api3Pool.EPOCH_LENGTH()))

### PR DESCRIPTION
**Pull request by BritikovKI**
Closes #182

### CODE REVIEW

File: packages/api3-voting/contracts/Api3Voting.sol [hunk 1]
```diff
@@ -97,6 +97,7 @@ contract Api3Voting is IForwarder, AragonApp {
         voteTime = _voteTime;
         // The pool acts as the MiniMe token
         api3Pool = IApi3Pool(_token);
+        require(_voteTime <= api3Pool.EPOCH_LENGTH(), ERROR_VOTING_LENGTH);
```
Comment (bbenligiray) [inline packages/api3-voting/contracts/Api3Voting.sol:100]: Should be exactly equal

File: packages/api3-voting/contracts/Api3Voting.sol [hunk 2]
```diff
@@ -33,6 +33,7 @@ contract Api3Voting is IForwarder, AragonApp {
     string private constant ERROR_CAN_NOT_EXECUTE = "VOTING_CAN_NOT_EXECUTE";
     string private constant ERROR_CAN_NOT_FORWARD = "VOTING_CAN_NOT_FORWARD";
     string private constant ERROR_NO_VOTING_POWER = "VOTING_NO_VOTING_POWER";
+    string private constant ERROR_VOTING_LENGTH = "VOTING_CANNOT_BE_LONGER_THEN_API3POOL_EPOCH";
```
Comment (bbenligiray) [inline packages/api3-voting/contracts/Api3Voting.sol:36]: `VOTING_NOT_EQUAL_TO_POOL_EPOCH`

**Review by bbenligiray — CHANGES_REQUESTED** [review]

File: packages/api3-voting/contracts/Api3Voting.sol (hunk 1 above)
Comment (BritikovKI) [reply packages/api3-voting/contracts/Api3Voting.sol]: Then I think we shouldn't create this variable anyway, it will be the same for all different votes

Comment (BritikovKI) [reply packages/api3-voting/contracts/Api3Voting.sol]: Will this refactoring be ok?

Comment (bbenligiray) [reply packages/api3-voting/contracts/Api3Voting.sol]: I just wanted to keep the `initialize()` interface identical to the original `Voting.sol`.

Ideally we should just go
```
voteTime = api3Pool.EPOCH_LENGTH();
```
Is there a reason why we can't do this?

Comment (BritikovKI) [reply packages/api3-voting/contracts/Api3Voting.sol]: We can do this, do you want to leave _voteTime as a parameter that we are passing? And do we need to check it if you want to?

**Comment by bbenligiray** [comment]
Don't you also need to change
https://github.com/api3dao/api3-dao/blob/935d845ed4e767314a383039563c3ff19d8c1981/packages/api3-voting/contracts/Api3Voting.sol#L86
and the deployment scripts, tests, etc.?

**Comment by BritikovKI** [comment]
Yep, I will need to, but I think this may be done in half an hour max, because the only thing we need to change is a function call, which is pretty easy to do inside a project

**Comment by bbenligiray** [comment]
Okay, but I think it makes sense to that too before the merge

**Comment by BritikovKI** [comment]
FInished

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**